dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java

```diff
@@ -352,7 +352,6 @@ public Token addGoogleToken(@ApiParam(hidden = true) @Auth Optional<User> authUs
                 user = new User();
                 // Pull user information from Google
                 GoogleHelper.updateUserFromGoogleUserinfoplus(userinfo, user);
-                user.setUsername(userinfo.getEmail());
```

Comment:
try keeping this one and deleting other one in updateUserFromGoogleUserinfoplus and see if it still works.

Reply:
`GoogleHelper.updateUserFromGoogleUserinfoplus` already invoked  `user.setUsername(userinfo.getEmail())`. I just removed this line because it seemed redundant.

I'm now calling `GoogleHelper.updateUserFromGoogleUserinfoplus` from SimpleAuthenticator and relying on the username being set there. If I remove the line from `updateUserFromGoogleUserinfoplus`, then I would have to explicitly set the username after invoking the function in two other places.

Can you please elaborate?

Thanks.

Reply:
the updateUserFromGoogleUserinfoplus was intended to update the user's metadata (profile and the avatarURL) only and is called in quite a few places.  The user's username should only be set once (when a new user is created).  Right now, if you have a GitHub account, and then link a Google account to it and update metadata, your username suddenly changes.

so setting the username should be removed from the updateUserFromGoogleUserinfoplus

Reply:
Got it, will update shortly.